src/transformers/models/qwen2_vl/modeling_qwen2_vl.py

```diff
@@ -1860,3 +1861,173 @@ def prepare_inputs_for_generation(
         return model_inputs
+
+
+@add_start_docstrings(
+    """(Inspired by the LlamaForSequenceClassification)
+    The Qwen2-VL transformer with a sequence classification head on top (linear layer).
+    Copied the Qwen2VLForConditionalGenerations forward method, but with modification such that it supports sequence classification using the last token embedding instead.
+
+    [`Qwen2VLForSequenceClassification`] uses the last token in order to do the classification, as other causal models
+    (e.g. GPT-2) do.
+
+    Since it does classification on the last token, it requires to know the position of the last token. If a
+    `pad_token_id` is defined in the configuration, it finds the last token that is not a padding token in each row. If
+    no `pad_token_id` is defined, it simply takes the last value in each row of the batch. Since it cannot guess the
+    padding tokens when `inputs_embeds` are passed instead of `input_ids`, it does the same (take the last value in
+    each row of the batch).
+    """,
+    QWEN2VL_START_DOCSTRING,
+)
+class Qwen2VLForSequenceClassification(Qwen2VLPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        self.num_labels = config.num_labels
+        self.visual = Qwen2VisionTransformerPretrainedModel._from_config(
+            config.vision_config, attn_implementation=config._attn_implementation
+        )
+        self.model = Qwen2VLModel(config)
+        self.vocab_size = config.vocab_size
+        self.classification_head = nn.Linear(config.hidden_size, self.num_labels, bias=False)
+        self.padding_side = "left"  # set it to left by default, user can use setter to change padding_sides
+
+        # Initialize weights and apply final processing
+        self.post_init()
+
+    def get_input_embeddings(self):
+        return self.model.embed_tokens
+
+    def set_input_embeddings(self, value):
+        self.model.embed_tokens = value
+
+    @add_start_docstrings_to_model_forward(QWEN2_VL_INPUTS_DOCSTRING)
+    def forward(
+        self,
+        input_ids: torch.LongTensor = None,
+        attention_mask: Optional[torch.Tensor] = None,
+        position_ids: Optional[torch.LongTensor] = None,
+        past_key_values: Optional[List[torch.FloatTensor]] = None,
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+        labels: Optional[torch.LongTensor] = None,
+        use_cache: Optional[bool] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        pixel_values: Optional[torch.Tensor] = None,
+        pixel_values_videos: Optional[torch.FloatTensor] = None,
+        image_grid_thw: Optional[torch.LongTensor] = None,
+        video_grid_thw: Optional[torch.LongTensor] = None,
+        rope_deltas: Optional[torch.LongTensor] = None,
+    ) -> Union[Tuple, SequenceClassifierOutputWithPast]:
+        r"""(copied from LlamaForSequenceClassification)
```

Comment:
let's not add (copied from ...) in docstring. In case the model code is 100% identical to Llama, we can add above the class/method `@ Copied from transformers.models.llama.modeling_llama.class_name.method_name_optionally`